### PR TITLE
[wip] poc: add deprecated information to generated docs

### PR DESCRIPTION
**Pull request by garethjevans**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by hferentschik** [comment]
@garethjevans, seems the formatting is off. `make fmt` might help.

Also, can we open an issue describing what this is about?
Is it about making sure that commands which are marked as deprecated in `deprecated.go` are also marked in the reference docs  - https://jenkins-x.io/docs/reference/commands/?

**Comment by garethjevans** [comment]
The markdown doc generator, does not generate docs for deprecated (or hidden) commands.  It would make sense that whilst the command is deprecated but still in use `jx install` is a good example, that we generate docs for them, but clearly mark them as deprecated.

File: pkg/cmd/deprecation/deprecated.go [hunk 1]
```diff
@@ -8,84 +8,85 @@ import (
...
+	Date        string
+	Info        string
 }
 
 // DeprecateCommands runs recursively over all commands and set the deprecation message
 // on every command defined the deprecated commands map.
 func DeprecateCommands(cmd *cobra.Command) {
 	path := commandPath(cmd)
-	if deprecation, ok := deprecatedCommands[path]; ok {
-		cmd.Deprecated = deprecationMessage(deprecation)
```
Comment (garethjevans) [inline pkg/cmd/deprecation/deprecated.go:88]: As you can see, we are no longer marking the cmd as "Deprecated", but updating the LongExample.  This is not ideal as it won't display when the command is run, just when help is called

**Pull request closed without merging**